### PR TITLE
add __eq__ to Device

### PR DESCRIPTION
**Pull request by mtar**
## Description

This PR overloads the '==' operator such that devices can be compared _locally_.

Issue/s resolved: #1062 

## Changes proposed:
- add `__eq__` operator

## Type of change

- New feature

## Memory requirements

n/a

## Performance

n/a

## Due Diligence
- [x] All split configurations tested
- [x] Multiple dtypes tested in relevant functions
- [x] Documentation updated (if needed)
- [x] Title of PR is suitable for corresponding CHANGELOG entry

#### Does this change modify the behaviour of other functions? If so, which?
no

### CODE REVIEW

File: heat/core/devices.py [hunk 1]
```diff
@@ -74,6 +74,31 @@ def __str__(self) -> str:
...
+
+        Parameters
+        ----------
+        other : Any
+            The object to compare with
+        """
+        if isinstance(other, Device):
+            return (
+                True
+                if (self.device_type == other.device_type and self.device_id == other.device_id)
```
Comment (Markus-Goetz) [inline heat/core/devices.py:89]: You can shorten this to a return of the condition, i.e. return self.device_type == other...

File: heat/core/devices.py [hunk 2]
```diff
@@ -74,6 +74,31 @@ def __str__(self) -> str:
...
+                else False
+            )
+
+        if isinstance(other, torch.device):
+            return (
+                True
+                if (self.device_type == other.type and self.device_id == other.index)
```
Comment (Markus-Goetz) [inline heat/core/devices.py:96]: Can equally be shortened, see above

**Review by Markus-Goetz — CHANGES_REQUESTED** [review]

**Review by ClaudiaComito — APPROVED** [review]
Great, thanks @mtar!
